aspnetcore/web-api/error-handling.md

```diff
@@ -0,0 +1,230 @@
+
+1. In `Startup.Configure`, invoke `UseExceptionHandler` to use the middleware:
+
+    ```csharp
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        if (env.IsDevelopment())
+        {
+            app.UseDeveloperExceptionPage();
+        }
+        else
+        {
+            app.UseExceptionHandler("/error");
+        }
+    }
+    ```
+
+1. Configure a controller action to respond to the `/error` route:
+
+    ```csharp
+    [ApiController]
+    public class ErrorController : ControllerBase
+    {
+        [Route("/error")]
+        public IActionResult Error() => Problem();
+    }
+    ```
+
+The preceding `Error` action sends an [RFC7807](https://tools.ietf.org/html/rfc7807)-compliant payload to the client.
+
+The Exception Handling Middleware can also provide more detailed content-negotiated output in the local development environment. Use the following steps to produce a consistent payload format across development and production environments:
+
+1. In `Startup.Configure`, register environment-specific Exception Handling Middleware instances:
+
+    ```csharp
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        if (env.IsDevelopment())
+        {
+            app.UseExceptionHandler("/error-local-development");
+        }
+        else
+        {
+            app.UseExceptionHandler("/error");
+        }
+    }
+    ```
+
+    In the preceding code, the middleware is registered with:
+
+    * A route of `/error-local-development` in the Development environment.
+    * A route of `/error` in environments that aren't Development.
+    
+1. Apply attribute routing to controller actions:
+
+    ```csharp
+    [ApiController]
+    public class ErrorController : ControllerBase
+    {
+        [Route("/error-local-development")]
+        public IActionResult ErrorLocalDevelopment(
+            [FromServices] IWebHostEnvironment webHostEnvironment)
+        {
+            if (!webHostEnvironment.IsDevelopment())
+            {
+                throw new InvalidOperationException(
+                    "This shouldn't be invoked in non-development environments.");
+            }
+    
+            var context = HttpContext.Features.Get<IExceptionHandlerFeature>();
+            return Problem(
+                detail: context.Error.StackTrace,
+                title: context.Error.Message);
+        }
+    
+        [Route("/error")]
+        public IActionResult Error() => Problem();
+    }
+    ```
+
+## Use exceptions to modify the response
+
+You may want to modify the contents of the response from outside of the controller. In ASP.NET 4.x Web API, one way to do this was using the `HttpResponseException` type. ASP.NET Core doesn't include an equivalent type. Support for `HttpResponseException` can be added with the following steps:
```

Comment:
Repo rules push back usually on use of 2nd person (but MS style is ok with it and it almost always leads to passive voice, which I'm not a big fan of). Anyway... if you want to factor it out (and I add the API link here) ...

```suggestion
The contents of the response can be modified from outside of the controller. In ASP.NET 4.x Web API, one way to do this was using the <xref:System.Web.Http.HttpResponseException> type. ASP.NET Core doesn't include an equivalent type. Support for `HttpResponseException` can be added with the following steps:
```

❓ Is that supposed to be "Web API" or "web API"?